### PR TITLE
BugFix - Adds check to onFocusChanged unfocusing when setting focus causing loop

### PR DESCRIPTION
**Pull request by NicolasChampagne**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

iOS Infinite loop

## What is the new behavior?

BugFix - Adds check to onFocusChanged unfocusing when setting focus causing loop

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):

https://nventive.visualstudio.com/Umbrella/_workitems/edit/157061

### CODE REVIEW

**Comment by jeromelaban** [comment]
Which platform does this fix ? Can you write a UI Test that creates the scenario ?

File: src/SamplesApp/SamplesApp.UITests/UnoSamples_Tests.Keyboard.cs [hunk 1]
```diff
@@ -275,6 +275,16 @@ public void TextBox_TextChanged()
 			;
 		}
 
+		[Test]
+		public void TextBox_PageLoadedTest()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.TextBoxTests.TextBox_TextChanged");
```
Comment (davidjohnoliver) [inline src/SamplesApp/SamplesApp.UITests/UnoSamples_Tests.Keyboard.cs]: did you confirm that without the fix this page crashes?

Comment (NicolasChampagne) [reply src/SamplesApp/SamplesApp.UITests/UnoSamples_Tests.Keyboard.cs]: Yes this page crashes without the fix, on iOS

**Comment by davidjohnoliver** [comment]
update release notes

**Comment by MakanWG** [comment]
There is also an issue on Android. It does not make the apps crash but it prevents TextBoxes to be unfocused, preventing the manual dismissal of the virtual keyboard.

**Pull request merged**